src/OrleansAWSUtils/Membership/DynamoDBMembershipTable.cs

```diff
@@ -0,0 +1,350 @@
+
+                MembershipTableData data = Convert(records);
+                if (logger.IsVerbose2) logger.Verbose2("ReadAll Table=" + Environment.NewLine + "{0}", data.ToString());
+
+                return data;
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.MembershipBase,
+                    $"Intermediate error reading all silo entries {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<bool> InsertRow(MembershipEntry entry, TableVersion tableVersion)
+        {
+            try
+            {
+                if (logger.IsVerbose) logger.Verbose("InsertRow entry = {0}", entry.ToFullString());
+                var tableEntry = Convert(entry);
+
+                bool result;
+
+                try
+                {
+                    var expression = $"attribute_not_exists({SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME}) AND attribute_not_exists({SiloInstanceRecord.SILO_IDENTITY_PROPERTY_NAME})";
+
+                    await storage.PutEntryAsync(TABLE_NAME_DEFAULT_VALUE, tableEntry.GetFields(true), expression);
+                    result = true;
+                }
+                catch (ConditionalCheckFailedException)
+                {
+                    result = false;
+                }
+
+                if (result == false)
+                    logger.Warn(ErrorCode.MembershipBase,
+                        $"Insert failed due to contention on the table. Will retry. Entry {entry.ToFullString()}");
+                return result;
+            }
+            catch (Exception exc)
+            {
+                logger.Warn(ErrorCode.MembershipBase,
+                    $"Intermediate error inserting entry {entry.ToFullString()} to the table {TABLE_NAME_DEFAULT_VALUE}.", exc);
+                throw;
+            }
+        }
+
+        public async Task<bool> UpdateRow(MembershipEntry entry, string etag, TableVersion tableVersion)
+        {
+            try
+            {
+                if (logger.IsVerbose) logger.Verbose("UpdateRow entry = {0}, etag = {1}", entry.ToFullString(), etag);
+                var siloEntry = Convert(entry);
+                int currentEtag = 0;
+                int.TryParse(etag, out currentEtag);
+                siloEntry.ETag = currentEtag;
+                siloEntry.ETag++;
+
+                bool result;
+
+                try
+                {
+                    var conditionalValues = new Dictionary<string, AttributeValue> { { CURRENT_ETAG_ALIAS, new AttributeValue { N = etag } } };
+                    var expression = $"{SiloInstanceRecord.ETAG_PROPERTY_NAME} = {CURRENT_ETAG_ALIAS}";
+                    await storage.UpsertEntryAsync(TABLE_NAME_DEFAULT_VALUE, siloEntry.GetKeys(),
```

Comment:
I agree with you that it can be dangerous but, if I do the way you are asking, it will add a bunch of boilerplate non-reusable code just as it was in Azure Table Storage provider. I'll explain why in another comment here.